src/m365/spo/commands/site/site-remove.spec.ts

```diff
@@ -81,1383 +104,193 @@ describe(commands.SITE_REMOVE, () => {
+        return;
       }
-
       throw 'Invalid request';
     });
 
-    sinon.stub(global, 'setTimeout').callsFake((fn) => {
-      fn();
-      return {} as any;
-    });
+    const deleteStub = sinon.stub(request, 'delete').resolves();
 
-    await command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/demositeGrouped', debug: true, verbose: true, force: true, wait: true } });
-    assert(loggerLogToStderrSpy.called);
+    await command.action(logger, { options: { url: siteUrl, fromRecycleBin: true, verbose: true } });
+    assert(deleteStub.notCalled);
   });
 
-  it('does not remove the site - Groupified Site, bad request (getting the site group)', async () => {
-    sinon.stub(request, 'post').callsFake(async (opts) => {
-      if ((opts.url as string).indexOf(`/_vti_bin/client.svc/ProcessQuery`) > -1) {
-        if (opts.data === `<Request xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009" AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}"><Actions><ObjectPath Id="4" ObjectPathId="3"/><Query Id="5" ObjectPathId="3"><Query SelectAllProperties="false"><Properties><Property Name="GroupId" ScalarProperty="true"/></Properties></Query></Query></Actions><ObjectPaths><Constructor Id="1" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}"/><Method Id="3" ParentId="1" Name="GetSitePropertiesByUrl"><Parameters><Parameter Type="String">https://contoso.sharepoint.com/sites/demositeGrouped</Parameter></Parameters></Method></ObjectPaths></Request>`) {
-          return JSON.stringify([
-            {
-              "SchemaVersion": "15.0.0.0",
-              "LibraryVersion": "16.0.20530.12001",
-              "ErrorInfo": null,
-              "TraceCorrelationId": "10f1829f-d000-0000-5962-1110d33e2cf2"
-            },
-            4,
-            {
-              "IsNull": false
-            },
-            5,
-            {
-              "_ObjectType_": "Microsoft.Online.SharePoint.TenantAdministration.SiteProperties",
-              "_ObjectIdentity_": "10f1829f-d000-0000-5962-1110d33e2cf2|908bed80-a04a-4433-b4a0-883d9847d110:095efa67-57fa-40c7-b7cc-e96dc3e5780c\nSiteProperties\nhttps%3a%2f%2fcontoso.sharepoint.com%2fsites%2fdemosite",
-              "GroupId": "/Guid(58587cc9-560c-4adb-a849-e669bd37c5f8)/"
-            }
-          ]);
-        }
-      }
 
-      throw 'Invalid request';
-    });
-
-    sinon.stub(request, 'get').callsFake(async (opts) => {
-      if (opts.url === 'https://graph.microsoft.com/v1.0/groups/58587cc9-560c-4adb-a849-e669bd37c5f8') {
-        throw {
-          response: {
-            status: 400,
-            data: {
-              error: { 'odata.error': { message: { value: "Invalid object identifier '0z013f997-c1aa-424d-a530-c4e3a05de704'." } } }
-            }
-          }
-        };
+  it('throws error if site is not found', async () => {
+    sinon.stub(odata, 'getAllItems').callsFake(async (url) => {
```

Comment:
In this case, we can just say `.resolves([])`, without the `callsFake`. In this test we don't really care if the URL is correct. That will be checked at other places.